vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/SiteMapParser.java

```diff
@@ -59,7 +70,34 @@ private Collection<SiteMapURL> parse(URI siteMapUrl, crawlercommons.sitemaps.Sit
         try
         {
             HttpClientContext context = new HttpClientContext();
-            HttpResponse response = httpClient.doHttpGet(siteMapUrl, context);
+            HttpGet request = new HttpGet(siteMapUrl);
+            URIAuthority currentAuthority = request.getAuthority();
+            String authorityUserInfo = currentAuthority.getUserInfo();
+            if (authorityUserInfo != null)
+            {
+                URIAuthority newAuthority = new URIAuthority(null, currentAuthority.getHostName(),
+                        currentAuthority.getPort());
+                request.setAuthority(newAuthority);
+
+                UriUtils.UserInfo userInfo = UriUtils.parseUserInfo(URLDecoder.decode(authorityUserInfo,
+                        StandardCharsets.UTF_8));
+                Credentials credentials = new UsernamePasswordCredentials(userInfo.user(),
+                        userInfo.password().toCharArray());
+                BasicScheme scheme = new BasicScheme();
+                scheme.initPreemptive(credentials);
```

Comment:
Do you really need **preemptive** basic auth?

Reply:
changed to CredentialsProvider